File: solutions/_1838_frequency_of_the_most_frequent_element/solution.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maxFrequency = function (nums, k) {
  nums.sort((a, b) => a - b);
  let l = 0;
  let r = 0;
  let sum = 0;
  let max = 0;

  while (r < nums.length) {
    sum += nums[r];
    while (sum + k < nums[r] * (r - l + 1)) {
      sum -= nums[l];
      l++;
    }
    max = Math.max(max, r - l + 1);
    r++;
  }

  return max;
};

module.exports = { maxFrequency };

/*
https://leetcode.com/problems/frequency-of-the-most-frequent-element/
*/

File: solutions/_1838_frequency_of_the_most_frequent_element/solution.java

package _1838_frequency_of_the_most_frequent_element;

import java.util.Arrays;

class Solution {
    public int maxFrequency(int[] nums, int k) {
        Arrays.sort(nums);
        int max = 0;
        long sum = 0;

        for (int l = 0, r = 0; r < nums.length; ++r) {
            sum += nums[r];
            while (sum + k < (long) nums[r] * (r - l + 1)) {
                sum -= nums[l];
                l++;
            }
            max = Math.max(max, r - l + 1);
        }

        return max;
    }
}

/*
 * https://leetcode.com/problems/frequency-of-the-most-frequent-element/
 */